### PR TITLE
Modify API to return info about auto-awarded STEAM badges

### PR DESCRIPTION
**Pull request by ghost**
POST /v2/user/badge/<shortname> and POST /v2/claim will now include an autoAwardedBadges array in their response, which contains the shortnames of any auto-awarded steam badges that resulted from the badge award.

This addresses https://github.com/mozilla/openbadger/issues/191 and will support https://github.com/mozilla/openbadger/issues/197.

### CODE REVIEW

File: routes/api.js [hunk 1]
```diff
@@ -336,14 +336,25 @@ function tryAwardingBadge(opts, res, successCb) {
         user: email,
       });
 
-    var success = res.send.bind(res, 200, {
-      status: 'ok',
-      url: instance.absoluteUrl('assertion'),
-    });
+    function getShortNameFromInstance(instance, cb) {
+      Badge.findById(instance.badge, function (err, badge) {
```
Comment (toolness) [inline routes/api.js]: I think the more common idiom we've been using is `instance.populate('badge', function(err) { ... })`, but this works too I guess.

**Comment by toolness** [comment]
Cool, and nice test!

Can you mention the new behavior in `docs/api.md`?

File: routes/api.js (hunk 1 above)
Comment (ghost) [inline routes/api.js]: Ah, a fine tip.  I shall change this.

**Comment by ghost** [comment]
API docs have been updated!

**Pull request merged**